projects/com.oracle.truffle.llvm.parser.impl/src/com/oracle/truffle/llvm/parser/impl/LLVMVisitor.java

```diff
@@ -392,8 +408,29 @@ private void allocateGlobals(List<EObject> objects) {
                     LLVMNode functionCall = factoryFacade.createFunctionCall(loadedFunction, argNodes, LLVMBaseType.VOID);
                     globalVarNodes.add(functionCall);
                 }
-            } else if (globalVar.getName().equals("llvm.global_dtors")) {
-                throw new AssertionError("destructors not yet supported!");
+            } else if (globalVar.getName().equals("@llvm.global_dtors")) {
+                ResolvedArrayType type = (ResolvedArrayType) typeResolver.resolve(globalVar.getType());
+                int size = type.getSize();
+                Object allocGlobalVariable = findOrAllocateGlobal(globalVar);
+                ResolvedType structType = type.getContainedType(0);
+                int structSize = LLVMTypeHelper.getByteSize(structType);
+                for (int i = 0; i < size; i++) {
+                    LLVMExpressionNode globalVarAddress = factoryFacade.createLiteral(allocGlobalVariable, LLVMBaseType.ADDRESS);
```

Comment:
Could you extract the shared code between constructors and destructors into a method?